v2/paths/interfaces.yaml

```diff
@@ -0,0 +1,132 @@
+# SPDX-License-Identifier: GPL-3.0
+# SPDX-FileCopyrightText: Canonical Ltd
+
+get:
+  tags:
+    - Interfaces
+    - OpenAccess
+    - Synchronous
+  summary: Get available interfaces
+  description: |-
+    Retrieves the available interfaces and their associated metadata.
+  operationId: getInterfaces
+  security: []
+  parameters:
+    - name: select
+      in: query
+      description: |-
+        Set to all to retrieve all interfaces, or connected to only return
+        connected interfaces (if this parameter is omitted then the call returns
+        the legacy format that should be no longer used).
+      schema:
+        type: string
+        enum:
+          - all
+          - connected
+    - name: slots
+      in: query
+      description: If set to true, slot information will be returned.
+      schema:
+        type: boolean
+    - name: plugs
+      in: query
+      description: If set to true, plug information will be returned.
+      schema:
+        type: boolean
+    - name: doc
+      in: query
+      description: If set to true, interface documentation will be returned.
+      schema:
+        type: boolean
+    - name: names
+      in: query
+      description: |-
+        Interfaces that match the listy of comma-seperated names will be returned.
+        The parameter matches against the name of the interface, not the name of
+        snaps, plugs, or slots.
+      schema:
+        type: string
+        example:
+          - content
+  responses:
+    200:
+      description: |-   # TODO Investigate whether discriminator can be used to map input parameter to oneOf result
+        A synchronous response containing an array of plugs and slots.
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              status-code:
+                type: integer
+                example: 200
+              status:
+                type: string
+                example: OK
+              type:
+                type: string
+                example: sync
+              result:
+                oneOf:
+                  - ref$: '../components/schemas/InterfaceArray.yaml'
+                  - ref$: '../components/schemas/InterfaceObject.yaml'
+    4XX:
+      $ref: '../components/responses/InternalError.yaml'
+
+post:
+  tags:
+    - Asynchronous
+    - AuthenticationRequired
+    - Interfaces
+  summary: Issue an action to the interface system
+  description: |-
+    Retrieves the available interfaces and their associated metadata.
```

Comment:
I think this was copied from the GET section, POST is used for connecting/disconnecting